### PR TITLE
Add deposited usd to pools. Fix superfluous rpc calls

### PR DESCRIPTION
**Pull request by memacius32**
Adding deposited $ usd on pools, also refactored the way we were fetching user's pools data.

- Added getTokenPrice hook. It uses useUsdcPrice hook and if it doesn't work, it simulates a swap to get the price.
- Added customSwap as opt parameter in useDerivedSwapInfo.
- Fixed useUsdcPrice and renamed it's pangolin/avax variables in favor of our ecosystem ones (didn't change the logic except checking if the token is USDT and some minor fixes).
-  Removed logic un useTrackedTokenPairs where it generated pairs for each posible token on user list's, and performed balances call on each one of them. Now we only fetch info on trisolaris defined pools, plus user's imported pools.

--

Main files to check out are :

- src/components/PositionCard/index.tsx
- src/hooks/useGetTokenPrice.ts
- src/hooks/useUSDCPrice.ts
- src/pages/Pool/index.tsx
- src/state/swap/hooks.ts
- src/state/user/hooks.tsx

The rest of the files are just locale changes or reorganizing stuff. i.e.: Separating Full position card and minimal position card into two separate files, and also separating styles files.

Preview:
Screen Shot 2022-03-13 at 16 31 15

### CODE REVIEW

**Comment by rej156** [comment]
Could we add a " : " (colon symbol) to suffix the Your deposits copy?

PR lgtm on first glance

**Comment by memacius32** [comment]
> Could we add a " : " (colon symbol) to suffix the Your deposits copy?
> 
> PR lgtm on first glance

missed it! fixed now + added locales

File: src/hooks/useUSDCPrice.ts [hunk 1]
```diff
@@ -47,34 +53,44 @@ export default function useUSDCPrice(currency?: Currency): Price | undefined {
...
         : BIG_INT_ZERO
 
     // all other tokens
     // first try the usdc pair
-    if (usdcPairState === PairState.EXISTS && usdcPair && usdcPair.reserveOf(USDC).greaterThan(avaxPairAVAXUSDCValue)) {
+    if (
+      usdcPairState === PairState.EXISTS &&
+      usdcPair &&
+      usdcPair.reserveOf(USDC).greaterThan(currencyEthPairWETHUSDCValue)
+    ) {
```
Comment (specialist0x) [inline src/hooks/useUSDCPrice.ts:74]:
```suggestion
      usdcPair?.reserveOf(USDC).greaterThan(currencyEthPairWETHUSDCValue)
    ) {
```

File: src/pages/Pool/index.tsx [hunk 2]
```diff
@@ -1,85 +1,37 @@
...
 
-  // fetch the user's balances of all tracked V2 LP tokens
   const trackedTokenPairs = useTrackedTokenPairs()
-  const { t } = useTranslation()
+
   const tokenPairsWithLiquidityTokens = useMemo(
     () =>
       trackedTokenPairs.map(tokens => ({
-        liquidityToken: toV2LiquidityToken(tokens, chainId ? chainId : ChainId.AVALANCHE),
+        liquidityToken: toV2LiquidityToken(tokens, chainId ? chainId : ChainId.AURORA),
```
Comment (specialist0x) [inline src/pages/Pool/index.tsx:34]:
```suggestion
        liquidityToken: toV2LiquidityToken(tokens, chainId ?? ChainId.AURORA),
```

File: src/state/swap/hooks.ts [hunk 3]
```diff
@@ -129,13 +131,14 @@ export function useDerivedSwapInfo(): {
 
   const toggledVersion = useToggledVersion()
 
+  const swapData = customSwap ?? useSwapState()
```
Comment (specialist0x) [inline src/state/swap/hooks.ts:134]: Whoa didn't know you could use hooks conditionally like this

**Comment by specialist0x** [comment]
Looks great!

For the copy, can the text be updated to:

```
Pool Tokens:
Pooled TRI:
Pooled AURORA:
Deposited Amount:
Pool Share:
```

Additionally, for `Deposited Amount`, can that be trimmed to show only 2 decimal places?

**Pull request merged**